api/api/views/viewsets/mission.py

```diff
@@ -436,61 +439,70 @@ class MissionViewset(viewsets.ModelViewSet):  # pylint: disable=too-many-ancesto
         security=['Bearer'],
         tags=['mission'],
     )
+
+    def retrieve(self, request, *args, **kwargs):
+        cached_key = f'{self.CACHE_KEY_PREFIX}{kwargs["pk"]}'
+
+        mission = cache.get(cached_key)
+        if not mission:
+            mission = self.get_object()
+            serializer = self.get_serializer(mission)
+            mission = serializer.data
+            cache.set(cached_key, mission, self.CACHE_TIMEOUT)
+            return super().retrieve(request, *args, **kwargs)
+        return Response(mission)
+
+    def list(self, request, *args, **kwargs):
+        cached_key = f'{self.CACHE_KEY_PREFIX}list'
```

Comment:
this code will leak missions to users that should not have access to them. 
You could fix that by:

1) filter mission by checking if user should have [access to them](https://github.com/vrn-sh/erp/blob/next/api/api/models/mission.py#L114) (you can add the `@property` decorator if that helps)

2) append the team's id to the cache key to make sure each cached version is unique for each teams